shub/config.py

```diff
@@ -267,11 +269,37 @@ def get_target_conf(self, target, auth_required=True):
             apikey=apikey,
             stack=(self.stacks.get(proj['stack'], proj['stack'])
                    if 'stack' in proj else self.stacks.get('default')),
+            image=self._select_image_for_project(target, proj),
             requirements_file=self.requirements_file,
             version=self.get_version(),
             eggs=self.eggs,
         )
 
+    def _select_image_for_project(self, target, project):
+        """Helper to select image for a project (or its target).
+
+        The select logic is the following:
+        - image defined per project has highest priority,
+        - images section is marked as deprecated, but still in force:
+          if target is defined in images - use a corresponding image,
+        - if image is not defined for project or target, default image
+          should be used instead (shortcut to images->default),
```

Comment:
Slightly confused by this because if image is not configured stacks should be used